plugin/src/App/init.lua

```diff
@@ -358,6 +359,19 @@ function App:startSession()
 		twoWaySync = sessionOptions.twoWaySync,
 	})
 
+	self.cleanupPrecommit = serveSession.__reconciler:hookPrecommit(function(patch, instanceMap)
+		-- Build new tree for patch
+		self:setState({
+			patchTree = PatchTree.build(patch, instanceMap, { "Property", "Old", "New" }),
+		})
+	end)
+	self.cleanupPostcommit = serveSession.__reconciler:hookPostcommit(function(patch, instanceMap, unappliedPatch)
+		-- Update tree with unapplied metadata
+		self:setState({
+			patchTree = PatchTree.updateMetadata(self.state.patchTree, patch, instanceMap, unappliedPatch),
+		})
```

Comment:
Updating state based on previous state is not allowed without using setState callback.
```suggestion
		self:setState(function(prevState)
			return {
				patchTree = PatchTree.updateMetadata(prevState.patchTree, patch, instanceMap, unappliedPatch),
			}
		end)
```